app/models/settings/smtp.rb

```diff
@@ -14,14 +14,24 @@ class SMTP < Base
 
     class << self
       def settings
-        return sendgrid_settings if ENV["SENDGRID_API_KEY"].present?
+        if provided_minimum_settings?
+          custom_provider_settings
+        else
+          fallback_sendgrid_settings
+        end
+      end
 
-        keys.index_with { |k| public_send(k) }.symbolize_keys
+      def provided_minimum_settings?
+        address.present? && user_name.present? && password.present?
       end
 
       private
 
-      def sendgrid_settings
+      def custom_provider_settings
+        keys.index_with { |k| public_send(k) }.symbolize_keys
```

Comment:
Suggestion: this may be a bit clearer. It also uses core Ruby and no Rails extensions (`index_with` was added to Rails before the block for `to_h` was added to Ruby):

```suggestion
        keys.to_h { |k| [k.to_sym, public_send(k)] }
```

Reply:
Also, this gave me the idea to maybe expose this for all settings models as part of `Settings::Base`.

Reply:
PR is up, we can change this after merging to simply call `to_h`.

Reply:
Just FYI, the other PR has been merged now in case you want to update but we can always do that later too.

Reply:
Updated!